test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+	output := strings.Split(str, "rate ")
+	rate := strings.Split(output[1], "burst")
+	regex := regexp.MustCompile("[0-9]+")
+	outputStr := regex.FindAllString(rate[0], -1)
+	outputInt, err := strconv.ParseInt(outputStr[0], 10, 64)
+	bwInt, err := BwConvertInt64(bw)
+	if err != nil {
+		return err
+	}
+	if bwInt == outputInt {
+		logrus.Infof("Applied bandwidth: %dkbits equals tc qdisc rate: %dkbits", bwInt, outputInt)
+	} else {
+		logrus.Errorf("Applied bandwidth: %dkbits does not match the tc rate: %d ", bwInt, outputInt)
+		return errors.New("Applied bandwidth doe sot match the tc qdisc rate")
+	}
+	return nil
+}
+
+func (k *kubePod) checkConnection(c *container, ipaddr, protocol string, port int) error {
+	var protoStr string
+
+	if protocol == "udp" {
+		protoStr = "-u"
+	}
+
+	logrus.Infof("Checking connection from %s to ip %s on port %d", c, ipaddr, port)
+
+	out, err := k.exec(c.containerID, fmt.Sprintf("nc -z -n -v -w 1 %s %s %v", protoStr, ipaddr, port))
+	if err != nil && !strings.Contains(out, "open") {
+		logrus.Errorf("Connection from %v to ip %s on port %d FAILED", *c, ipaddr, port)
+	} else {
+		logrus.Infof("Connection from %v to ip %s on port %d SUCCEEDED", *c, ipaddr, port)
+	}
+
+	return err
+}
+
+func (k *kubePod) checkNoConnection(c *container, ipaddr, protocol string, port int) error {
+	logrus.Infof("Expecting connection to fail from %v to %s on port %d", c, ipaddr, port)
+
+	if err := k.checkConnection(c, ipaddr, protocol, port); err != nil {
+		return nil
+	}
+	return fmt.Errorf("Connection SUCCEEDED on port %d from %s from %v when it should have FAILED.", port, ipaddr, c)
+}
+
+func (k *kubePod) cleanupContainers() error {
+	if k.node.Name() == "k8master" {
+		logrus.Infof("Cleaning up containers on %s", k.node.Name())
+		k8sMaster.tbnode.RunCommand(fmt.Sprintf("kubectl delete pod"))
+	}
+	return nil
+}
+
+func (k *kubePod) startNetplugin(args string) error {
+	if k.node.Name() == "k8master" {
+		return nil
+	}
+	podName, err := getPodName("netplugin", k.node.Name())
+	if err != nil {
+		logrus.Errorf("pod not found: %+v", err)
+		return err
+	}
+
+	logrus.Infof("Starting netplugin on %s", k.node.Name())
+	startNetpluginCmd := k.node.suite.basicInfo.BinPath + `/netplugin -plugin-mode=kubernetes -vlan-if=` + k.node.suite.hostInfo.HostDataInterfaces + ` -cluster-store=` + k.node.suite.basicInfo.ClusterStore + ` ` + args + ` > ` + netpluginLogLocation + ` 2>&1`
+
+	return k.podExecBG(podName, startNetpluginCmd, "kube-system")
+}
+
+func (k *kubePod) stopNetplugin() error {
+	podName, err := getPodName("netplugin", k.node.Name())
+	if err != nil {
+		logrus.Errorf("pod not found: %+v", err)
+		return err
+	}
+
+	stopRestartCmd := `rm ` + netpluginRestartFile
+	k.exec(podName, stopRestartCmd, "kube-system")
+
+	logrus.Infof("Stopping netplugin on %s", k.node.Name())
+	killNetpluginCmd := `pkill netplugin`
+	_, err = k.exec(podName, killNetpluginCmd, "kube-system")
+	return err
+}
+
+func (k *kubePod) stopNetmaster() error {
```

Comment:
Yes, pod will still remain. The process inside the container is killed and started with custom arguments as necessary by test cases